cli/installer/install-azd.ps1

```diff
@@ -341,6 +345,16 @@ try {
 
     try {
         if (isLinuxOrMac) {
+            if ($IsMacOS -and (-not $SkipVerify)) {
+                Write-Verbose "Verifying signature of $binFilename" -Verbose:$Verbose
+                codesign -v "$tempFolder/decompress/$binFilename" *>$null
+                if (-not $?) {
```

Comment:
I'm not sure why, but when running `pwsh` on macOS, `$LASTEXITCODE` did not capture the exit code from running `codesign -v`, and I ended up switching to use `$?` instead.

Reply:
Is `codesign` an alias, perhaps? pwsh on Windows does that for `dir`, `ls`, etc., but of course the switches aren't the same. The one I always unalias in my `$profile` is `curl`. Ew. But `codesign -v` would map correctly. You might check. If it's aliased on Mac, I would make sure you use the standard utility in case behavior varies.

If it is an alias, just run prior: `Remove-Alias codesign -ErrorAction Ignore`

Reply:
`$lastexitcode` must be set differently on Mac then. That said, I don't see anything wrong with what you did. Just...interesting. `$?` is valid for cmdlets and native apps alike so it's definitely okay to use here.

Reply:
What is interesting is that it does work inside an interactive shell:

```posh
PS /Users/weilim/repos/azure-dev/cli/installer> codesign -dv -r- /usr/local/bin/azd                 /usr/local/bin/azd: code object is not signed at all                     
PS /Users/weilim/repos/azure-dev/cli/installer> $LASTEXITCODE
1
```

I'll try to get a minimal repro and log the issue for PowerShell

Reply:
Fixed the bug in the current PowerShell script. My faith in PowerShell is somewhat restored.

We were setting `$LASTEXITCODE=0` somewhere in the top of the script. This creates a script-scoped variable `$LASTEXITCODE` which was shadowing `$global:$LASTEXITCODE`.